### PR TITLE
Add latency numbers to increment tool

### PR DESCRIPTION
**Pull request by animesh2049**
Add query, mutation and total txn latency numbers to increment tool.
Set `CommitNow` to be true in mutation.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/3422)

### CODE REVIEW

File: dgraph/cmd/counter/increment.go [hunk 1]
```diff
@@ -74,6 +75,15 @@ func queryCounter(txn *dgo.Txn, pred string) (Counter, error) {
 	var counter Counter
 	query := fmt.Sprintf("{ q(func: has(%s)) { uid, val: %s }}", pred, pred)
 	resp, err := txn.Query(ctx, query)
+
+	// Total query latency is sum of encoding, parsing and processing
+	// latencies
```
Comment (mangalaman93) [inline dgraph/cmd/counter/increment.go]: could move up in the same line, and a full stop at the end

File: dgraph/cmd/counter/increment.go [hunk 2]
```diff
@@ -74,6 +75,15 @@ func queryCounter(txn *dgo.Txn, pred string) (Counter, error) {
...
+	queryLatency := resp.Latency.GetEncodingNs() +
+		resp.Latency.GetParsingNs() + resp.Latency.GetProcessingNs()
+	latency := strconv.FormatUint(queryLatency, 10) + "ns"
```
Comment (mangalaman93) [inline dgraph/cmd/counter/increment.go]: This seems unnecessary. You could just say
`latency := time.NanoSecond * queryLatency`

File: dgraph/cmd/counter/increment.go [hunk 3]
```diff
@@ -118,20 +128,31 @@ func process(dg *dgo.Dgraph, conf *viper.Viper) (Counter, error) {
...
-	_, err = txn.Mutate(context.Background(), &mu)
+	resp, err := txn.Mutate(context.Background(), &mu)
 	if err != nil {
 		return Counter{}, err
 	}
-	return counter, txn.Commit(context.Background())
+
+	mutationLatency := resp.Latency.GetProcessingNs() +
+		resp.Latency.GetParsingNs() + resp.Latency.GetEncodingNs()
+	latency := strconv.FormatUint(mutationLatency, 10) + "ns"
```
Comment (mangalaman93) [inline dgraph/cmd/counter/increment.go]: same here

**Comment by animesh2049** [comment]
Sample output
```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ time ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck
Query latency: 12.328646ms
Mutation latency: 15.267391ms
0515 12:33:19.082 Counter VAL: 84   [ Ts: 10159 ]
Total txn latency is: 29.289451ms

real	0m0.044s
user	0m0.008s
sys	0m0.016s
```

**Comment by animesh2049** [comment]
```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck --num=10
Query latency: 3.323626ms
Mutation latency: 14.69912ms
0515 12:54:04.699 Counter VAL: 95   [ Ts: 10181 ]
Query latency: 16.204665ms
Mutation latency: 15.557931ms
0515 12:54:04.732 Counter VAL: 96   [ Ts: 10183 ]
Query latency: 16.045075ms
Mutation latency: 15.72075ms
0515 12:54:04.764 Counter VAL: 97   [ Ts: 10185 ]
Query latency: 16.251635ms
Mutation latency: 15.58223ms
0515 12:54:04.797 Counter VAL: 98   [ Ts: 10187 ]
Query latency: 16.173984ms
Mutation latency: 15.570271ms
0515 12:54:04.829 Counter VAL: 99   [ Ts: 10189 ]
Query latency: 16.903767ms
Mutation latency: 14.962927ms
0515 12:54:04.862 Counter VAL: 100   [ Ts: 10191 ]
Query latency: 14.745019ms
Mutation latency: 15.479122ms
0515 12:54:04.894 Counter VAL: 101   [ Ts: 10193 ]
Query latency: 16.296664ms
Mutation latency: 15.51334ms
0515 12:54:04.927 Counter VAL: 102   [ Ts: 10195 ]
Query latency: 16.483921ms
Mutation latency: 15.268494ms
0515 12:54:04.959 Counter VAL: 103   [ Ts: 10197 ]
Query latency: 16.359744ms
Mutation latency: 15.5736ms
0515 12:54:04.992 Counter VAL: 104   [ Ts: 10199 ]
Total txn latency is: 312.788034ms
```

**Comment by animesh2049** [comment]
Output after changing format.
```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck --num=1
0516 08:20:30.571 Counter VAL: 250   [ Ts: 10491 ][ 4ms 14ms 20ms ]
Total: 21ms
```

```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck --num=10
0516 08:20:48.18  Counter VAL: 251   [ Ts: 10493 ][ 5ms 14ms 20ms ]
0516 08:20:48.213 Counter VAL: 252   [ Ts: 10495 ][ 16ms 15ms 32ms ]
0516 08:20:48.245 Counter VAL: 253   [ Ts: 10497 ][ 16ms 14ms 32ms ]
0516 08:20:48.278 Counter VAL: 254   [ Ts: 10499 ][ 16ms 14ms 32ms ]
0516 08:20:48.31  Counter VAL: 255   [ Ts: 10501 ][ 16ms 15ms 32ms ]
0516 08:20:48.343 Counter VAL: 256   [ Ts: 10503 ][ 16ms 14ms 32ms ]
0516 08:20:48.376 Counter VAL: 257   [ Ts: 10505 ][ 16ms 14ms 32ms ]
0516 08:20:48.408 Counter VAL: 258   [ Ts: 10507 ][ 16ms 14ms 32ms ]
0516 08:20:48.44  Counter VAL: 259   [ Ts: 10509 ][ 17ms 14ms 32ms ]
0516 08:20:48.473 Counter VAL: 260   [ Ts: 10511 ][ 16ms 14ms 32ms ]
Total: 314ms
```

**Comment by animesh2049** [comment]
Results after further modifications
```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck
0516 07:25:06.977 Counter VAL: 401   [ Ts: 50041 ][ 13ms 12ms 25ms 29ms ]
Total: 31ms
```

```
animesh@workstation:~/go/src/github.com/dgraph-io/dgraph/dgraph$ ./dgraph increment --alpha 127.0.0.1:9080 --pred=counter.val.healthcheck --num=10
0516 07:25:09.025 Counter VAL: 402   [ Ts: 50043 ][ 13ms 12ms 25ms 29ms ]
0516 07:25:09.061 Counter VAL: 403   [ Ts: 50045 ][ 21ms 12ms 33ms 35ms ]
0516 07:25:09.102 Counter VAL: 404   [ Ts: 50047 ][ 19ms 12ms 31ms 41ms ]
0516 07:25:09.137 Counter VAL: 405   [ Ts: 50049 ][ 21ms 12ms 33ms 35ms ]
0516 07:25:09.189 Counter VAL: 406   [ Ts: 50051 ][ 27ms 22ms 49ms 52ms ]
0516 07:25:09.222 Counter VAL: 407   [ Ts: 50053 ][ 19ms 12ms 31ms 33ms ]
0516 07:25:09.258 Counter VAL: 408   [ Ts: 50055 ][ 21ms 12ms 33ms 36ms ]
0516 07:25:09.294 Counter VAL: 409   [ Ts: 50057 ][ 21ms 12ms 32ms 36ms ]
0516 07:25:09.329 Counter VAL: 410   [ Ts: 50059 ][ 21ms 12ms 33ms 35ms ]
0516 07:25:09.364 Counter VAL: 411   [ Ts: 50061 ][ 21ms 12ms 33ms 36ms ]
Total: 369ms
```

**Pull request merged**